### DISCUSSION TITLE
How to use a templateRef with ndcDynamicContent.

### DISCUSSION THREAD

**Discussion by kotarCreative** [gund/ng-dynamic-component#507] [Q&A] [answered]
I am trying to use a templateRef to inject children into my ndc-dynamic component.

My code looks something like this:
```
<ng-template #child>
   ... my ng-content here...
</ng-template>
<ndc-dynamic [ndcDynamicComponent]="dynamicComponent" [ndcDynamicContent]="[[child]]"></ndc-dynamic>
```

Any help would be appreciated 😄

**Comment by gund** [comment, marked as answer]
I don't think you can render ng-templates directly as they are not renderable by default.
So you need to pre-render TemplateRef into some html node and then grab the ref to that node and pass it in as content:
```html
<ng-template #childTpl>
   ... my ng-content here...
</ng-template>
<div #childContent>
  <ng-container *ngTemplateOutlet="childTpl"></ng-container>
</div>
<ndc-dynamic [ndcDynamicComponent]="dynamicComponent" [ndcDynamicContent]="[[childContent.nativeElement]]"></ndc-dynamic>
```
But I would recommend to just pass a TemplateRef as an input to a dynamic component and then render it inside of your component.

**Reply by kotarCreative** [reply]
Thanks for the help!
